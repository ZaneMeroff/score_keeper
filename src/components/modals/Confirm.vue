<template>

 <q-dialog :dark="true" v-model="showModal" persistent>
    <q-card class="q-card">

      <q-card-section>
        <!-- header -->
        <div class="text q-mb-lg">{{ text }}</div>
          
        <q-btn-group flat class="q-btn-group">
          <!-- yes button -->
          <q-btn
            flat
            class="q-btn"
            color="green"
            size="md"
            :label="'yes'"
            @click="onYes"
          />
          <!-- no button -->
          <q-btn
            flat
            class="q-btn"
            color="red"
            size="md"
            :label="'no'"
            @click="onNo"
          />
        </q-btn-group>
      </q-card-section>
    </q-card>
  </q-dialog>

</template>

<script lang="ts">

// Decorators
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class Confirm extends Vue {
  
  @Prop(Function) private onYes!: Function
  @Prop(Function) private onNo!: Function
  @Prop(Boolean) private showModal!: boolean
  @Prop(String) private text!: string 
}
</script>

<style scoped>

  .q-card {
    background-color: #1e1e1e;
    width: 300px;
  }

  .q-btn-group {
    display: inline !important;
  }

  .q-btn {
    width: 50%;
  }

  .text {
    color: #FFF;
    font-size: 16px;
    text-align: center;
  }

</style>