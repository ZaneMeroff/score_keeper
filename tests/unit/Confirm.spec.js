import { createLocalVue, shallowMount } from '@vue/test-utils'
import Confirm from '@/components/modals/Confirm.vue'
import Quasar, * as All from 'quasar'

// ------------ Quasar setup ----------------

const localVue = createLocalVue()
localVue.use(Quasar, {components: All, directives: All, plugins: All})

// ------------------------------------------

describe('Confirm', () => {

  describe("snapshots", () => {

    it("should render with default html", () => {
      const propsData = {
        onYes: jest.fn(),
        onNo: jest.fn(),
        showModal: true,
        text: 'sample text'
      }
      const component = shallowMount(Confirm, { localVue, propsData })

      expect(component.element).toMatchSnapshot()
    })
  })

  describe('props', () => {

    describe('onYes', () => {
      
      it('should accept a function', () => {
        const propsData = {
          onYes: jest.fn(),
          onNo: jest.fn(),
          showModal: true,
          text: 'sample text'
        }
        const component = shallowMount(Confirm, { localVue, propsData })
  
        expect(component.vm.$props.onYes).toEqual(propsData.onYes)
      })
    })

    describe('onNo', () => {
      
      it('should accept a function', () => {
        const propsData = {
          onYes: jest.fn(),
          onNo: jest.fn(),
          showModal: true,
          text: 'sample text'
        }
        const component = shallowMount(Confirm, { localVue, propsData })
  
        expect(component.vm.$props.onNo).toEqual(propsData.onNo)
      })
    })

    describe('showModal', () => {
      
      it('should accept a boolean', () => {
        const propsData = {
          onYes: jest.fn(),
          onNo: jest.fn(),
          showModal: true,
          text: 'sample text'
        }
        const component = shallowMount(Confirm, { localVue, propsData })
  
        expect(component.vm.$props.showModal).toEqual(propsData.showModal)
      })
    })

    describe('text', () => {
      
      it('should accept a string', () => {
        const propsData = {
          onYes: jest.fn(),
          onNo: jest.fn(),
          showModal: true,
          text: 'sample text'
        }
        const component = shallowMount(Confirm, { localVue, propsData })
  
        expect(component.vm.$props.text).toEqual(propsData.text)
      })
    })
  })
})