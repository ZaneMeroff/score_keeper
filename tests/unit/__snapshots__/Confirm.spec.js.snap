// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm snapshots should render with default html 1`] = `
<q-dialog-stub
  persistent="true"
  position="standard"
  value="true"
>
  <q-card-stub
    class="q-card"
    dark="true"
    tag="div"
  >
    <q-card-section-stub
      tag="div"
    >
      <div
        class="text q-mb-lg"
        style="color: rgb(255, 255, 255);"
      >
        sample text
      </div>
       
      <q-btn-group-stub
        class="q-btn-group"
        flat="true"
      >
        <q-btn-stub
          align="center"
          class="q-btn"
          color="green"
          flat="true"
          label="yes"
          ripple="true"
          size="md"
        />
         
        <q-btn-stub
          align="center"
          class="q-btn"
          color="red"
          flat="true"
          label="no"
          ripple="true"
          size="md"
        />
      </q-btn-group-stub>
    </q-card-section-stub>
  </q-card-stub>
</q-dialog-stub>
`;
